crates/nu-cli/tests/completions/mod.rs

```diff
@@ -390,6 +390,34 @@ fn external_commands_only() {
     match_suggestions(&expected, &suggestions);
 }
 
+/// Disable external commands except for those start with `^`
+#[test]
+fn external_commands_disabled() {
+    let mut engine = new_external_engine();
+
+    let mut config = Config::default();
+    config.completions.external.enable = false;
+    engine.set_config(config);
+
+    let stack = nu_protocol::engine::Stack::new();
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+    let completion_str = "ls; ^sleep";
```

Comment:
Why the `ls;`?

Reply:
Not necessary, but it makes the case a little bit harder, to correctly detect `^` in that pipeline element.

Reply:
Oh I see, we don't want to just match lines beginning with `^`. Makes sense